### PR TITLE
[FEATURE] Enable self-initializing `ExpectColumnUniqueValueCountToBeBetween`

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Follow existing pattern to make `ExpectColumnUniqueValueCountToBeBetween` self-initializing
- Add an integration test to cover the behavior
- Add an Expectation utility to be able to easily extract the default profiler config from a given Expectation (thanks @alexsherstinsky !)

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/expectations/core/expect_column_values_to_be_between.py [hunk 1]
```diff
@@ -108,7 +108,7 @@ class ExpectColumnValuesToBeBetween(ColumnMapExpectation):
         "profiler_config",
     )
 
-    min_estimato_parameter_builder_config: ParameterBuilderConfig = (
+    min_estimator_parameter_builder_config: ParameterBuilderConfig = (
```
Comment (cdkini) [inline great_expectations/expectations/core/expect_column_values_to_be_between.py:111]: Minor typo - LMK if this should be moved to another PR.

File: great_expectations/expectations/core/expect_column_unique_value_count_to_be_between.py [hunk 2]
```diff
@@ -83,6 +95,73 @@ class ExpectColumnUniqueValueCountToBeBetween(ColumnExpectation):
     success_keys = (
         "min_value",
         "max_value",
+        "auto",
+        "profiler_config",
+    )
+
+    column_unique_values_range_estimator_parameter_builder_config: ParameterBuilderConfig = ParameterBuilderConfig(
+        module_name="great_expectations.rule_based_profiler.parameter_builder",
+        class_name="NumericMetricRangeMultiBatchParameterBuilder",
```
Comment (cdkini) [inline great_expectations/expectations/core/expect_column_unique_value_count_to_be_between.py:104]: ValueSet?

File: great_expectations/expectations/expectation.py [hunk 3]
```diff
@@ -2442,3 +2441,22 @@ def _format_map_output(
...
+
+    Returns:
+        The default profiler config within the target Expectation.
+        If not available, returns None.
+    """
+    expectation_impl = get_expectation_impl(expectation_name=expectation_type)
+    profiler_config: Optional[
+        RuleBasedProfilerConfig
+    ] = expectation_impl.default_kwarg_values.get("profiler_config")
+    return profiler_config
```
Comment (cdkini) [inline great_expectations/expectations/expectation.py:2426]: @alexsherstinsky I made some minor modifications to keep this simple but lmk if you'd like changes.

File: tests/integration/profiling/rule_based_profilers/test_profiler_user_workflows.py [hunk 4]
```diff
@@ -31,10 +34,23 @@
...
+@pytest.fixture
+def set_consistent_seed_within_expectation_default_profiler_config() -> Callable:
+    def _set_seed(expectation_type: str):
+        default_profiler: Optional[
+            RuleBasedProfilerConfig
+        ] = get_default_profiler_config_for_expectation_type(expectation_type)
+        assert default_profiler is not None and default_profiler.variables is not None
+        default_profiler.variables["bootstrap_random_seed"] = RANDOM_SEED
+
+    return _set_seed
```
Comment (cdkini) [inline tests/integration/profiling/rule_based_profilers/test_profiler_user_workflows.py:51]: @NathanFarmer am I doing this right? 😅

File: great_expectations/expectations/core/expect_column_values_to_be_between.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/expectations/core/expect_column_values_to_be_between.py:111]: 🙇

**Review by alexsherstinsky — APPROVED** [review]
LGTM -- beautiful!  Thanks!
